code/datums/mind.dm

```diff
@@ -227,7 +227,8 @@
 
 	enslaved_to = creator
 
-	current.faction = creator.faction.Copy()
+	current.faction |= creator.faction
+	creator.faction |= current.faction

```

Comment:
Due to kor's comment about cult golems' dominate not preventing monsters from harming the creator